src/main/java/net/neoforged/neoforge/transfer/ResourceHandlerUtil.java

```diff
@@ -163,95 +165,52 @@ public static <T extends IResource> int insertStacking(
-//     * @param filter       The filter to apply to the resources
-//     * @param index        The index that is being checked in the handler.
-//     * @param amount       the desired amount of the resource to extract.
-//     * @param transaction  The transaction context for the operation.
-//     *                     Passing in {@code null} will open a root transaction, whereas passing in a transaction will
-//     *                     allow you to make the final decision to commit based on the results of this method.
-//     * @param stackFactory A factory the given a resource of type {@code <R>} and an amount, a stack of type {@code <S>} can be created. The return is expected to be non-null and properly be the instanced empty value for a given resource.
-//     * @return a stack of type {@code <S>} typically in the form of an ResourceStack or as an example an ItemStack based on the factory provided
-//     */
-//    public static <R extends IResource, S> S extract(
-//            IResourceHandler<R> handler,
-//            Predicate<R> filter,
-//            int index,
-//            int amount,
-//            R defaultResource,
-//            @Nullable TransactionContext transaction,
-//            IStackFactory<R, S> stackFactory) {
-//        TransferPreconditions.checkNonNegative(amount);
-//        if (amount == 0) return stackFactory.create(defaultResource, 0);
-//
-//        R resource = handler.getResource(index);
-//        if (doesNotMatch(filter, resource))
-//            return stackFactory.create(defaultResource, 0);
-//        try (Transaction subTransaction = Transaction.open(transaction)) {
-//            int extract = handler.extract(resource, amount, subTransaction);
-//            if (extract == 0)
-//                return stackFactory.create(defaultResource, 0);
-//            subTransaction.commit();
-//            return stackFactory.create(resource, extract);
-//        }
-//    }
+    /**
+     * Extracts the first resource from an {@link IResourceHandler} that is not empty and matches the given filter.
+     *
+     * @param <T>         The type of resource handled by the handler
+     * @param handler     The {@link IResourceHandler} to extract the resource from. Can be {@code null}, which makes this method a no-op.
+     * @param filter      The filter to apply to the resources
+     * @param amount      The desired amount of the resource to extract
+     * @param transaction The transaction context for the operation.
+     *                    Passing in {@code null} will open a root transaction, whereas passing in a transaction will
+     *                    allow you to make the final decision to commit based on the results of this method.
+     * @return the resource and amount that was extracted or {@code null} if nothing was extracted
+     */
+    @Nullable
+    public static <T extends IResource> ResourceStack<T> extract(
```

Comment:
Do we want to call this `extractAny` to make it clear that the resource is not known ahead of time?

Reply:
Potentially extractFirst to match the Stream API's findFirst?